tower/src/util/combinators/try_with.rs

```diff
@@ -0,0 +1,74 @@
+use futures_util::future::{ready, Either, Ready};
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// Service returned by the [`try_with`] combinator.
```

Comment:
can we get a bit more in the docs here? The name isn't clear enough that we probably want to add some more docs.

Reply:
We can also link to the docs in the extension trait.

Reply:
This should link to the combinator that creates it, which is the authoritative source for the docs. It's also what users will probably see first. Since the combinator docs are linked, I don't think the type really needs any additional documentation --- it would just add the burden of having to maintain two separate versions of essentially the same documentation, keep them in sync, and so on. 

This is basically what std does for iterator combinators etc.